390/sophie_li.Rmd

```diff
@@ -0,0 +1,70 @@
+---
+title: "MP2 Visualization"
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+```{r}
+library(tidyverse)
+```
+
+```{r}
+stem_1 <- read_csv("../tree_main_census/data/census-csv-files/scbi.stem1.csv")
+stem_2 <- read_csv("../tree_main_census/data/census-csv-files/scbi.stem2.csv")
+stem_3 <- read_csv("../tree_main_census/data/census-csv-files/scbi.stem3.csv")
+spptable <- read_csv("../tree_main_census/data/census-csv-files/scbi.spptable.csv")
+
+full_data <- rbind(stem_1, stem_2, stem_3) %>%
+  mutate(ExactDate = lubridate::mdy(ExactDate),
+         year_month = lubridate::floor_date(ExactDate, "month"))
+```
+
+
+```{r}
+# checking NA distribution
+table(full_data$CensusID, is.na(as.numeric(full_data$dbh)))
+
+# checking date distribution
+table(full_data$CensusID, is.na(full_data$ExactDate))
+
+full_data %>% 
+  filter(!is.na(ExactDate)) %>% 
+  group_by(CensusID) %>% 
+  summarize(min_date = min(ExactDate),
+            max_date = max(ExactDate))
+
+# subsetting 2013 and 2018 data to plot 5-year difference
+dbh_diff <- stem_2 %>% 
+  rename(dbh_13 = dbh) %>% 
+  left_join(stem_3, by = c("treeID", "stemID", "sp")) %>% 
+  mutate(dbh_13 = as.numeric(dbh_13),
+         dbh_18 = as.numeric(dbh)) %>% 
+  filter(!is.na(dbh_13), !is.na(dbh_18), status.y == "A") %>% # filter for trees that are alive
+  group_by(treeID, stemID, sp) %>% 
+  summarize(dbh_diff = dbh_18 - dbh_13)
+  
+dbh_diff <- dbh_diff %>% 
+  left_join(spptable, by = c("sp")) %>% 
+  select(treeID, stemID, Family, dbh_diff)
+  
+```
+
+```{r}
+# Histogram and box plot
+
+ggplot(dbh_diff, aes(dbh_diff)) +
```

Comment:
Nice job with the plot! How about using "Growth" instead of "difference in" in the title tho? 

Any ideas on what you can do for a shiny app?

Reply:
I'm not sure yet, but I'm thinking of either adding interactivity by tree family, or adding an interactive dbh range and binwidth slider for the histogram.